app/javascript/App/components/NewsfeedPage/CoinListItem.js

```diff
@@ -1,33 +1,27 @@
 import React from 'react'
 import PercentageChange from '../PercentageChange'
 import Currency from '../Currency'
+import WatchButton from './WatchButton'
 
-export default ({ coin, setActiveEntity, activeEntity }) => {
-  let klass = 'pa3 bb b--light-gray flex justify-between items-center pointer'
-  if (
-    activeEntity &&
-    activeEntity.type === 'coin' &&
-    activeEntity.id === coin.get('id')
-  )
-    klass += ' bg-foam'
+export default (props) => {
```

Comment:
i think it's better to pass in just what we need. i don't like the idea of sending the ⛵️of props to descendants. let me know what you think!

Reply:
I definitely favor passing in everything, in past projects I spent (wasted) a lot of time on shuffling individual props around. But if you think that we'll see performance issues in future then we can of course. I guess for small, frequently rendered components like this it makes sense.

Reply:
here's a thread that discusses exactly this, albeit with very little fanfare: 

https://stackoverflow.com/questions/43532684/in-react-is-it-a-good-practice-to-pass-all-the-props-from-parent-to-child-compo